app/columns.tsx

```diff
@@ -6,40 +6,30 @@ import { PreviewLinkButton } from "@/components/ui/previewlinkbutton";
 import { ChevronsUpDown, ChevronDown, ChevronRight } from "lucide-react";
 import { PrettyDateRange } from "./dates";
 import { InfoPopover } from "@/components/ui/infopopover";
-
-function getSurfaceTagColor(template: string): string {
-  const surfaceTagColors: any = {
-    aboutwelcome: "bg-red-400",
-    defaultaboutwelcome: "bg-orange-400",
-    feature_callout: "bg-yellow-300",
-    infobar: "bg-lime-300",
-    milestone_message: "bg-green-400",
-    multi: "bg-teal-300",
-    pb_newtab: "bg-sky-400",
-    protections_panel: "bg-blue-500",
-    toast_notification: "bg-indigo-400",
-    toolbar_badge: "bg-purple-400",
-    spotlight: "bg-pink-400",
-    update_action: "bg-rose-400",
-  };
-
-  if (template in surfaceTagColors) {
-    return surfaceTagColors[template];
-  }
-
-  return "";
-}
+import { getSurfaceDataForTemplate } from "@/lib/messageUtils";
 
 function SurfaceTag(template: string, surface: string) {
-  let surfaceTagClassName = "";
-  if (template !== "none") {
-    surfaceTagClassName =
-      "px-2 py-1 inline rounded-md " + getSurfaceTagColor(template);
-  }
+  const { tagColor, docs } = getSurfaceDataForTemplate(template);
+  const surfaceTagClassName =
+    "text-xs/[180%] text-nowrap px-2 py-1 inline rounded-md " + tagColor;
```

Comment:
Minor nit for clarity - now that we know this works, we can tidy it up a bit by putting both the re-used classes into variables, and have another `const` for the anchor tag:
```
const anchorTagClassName = "visited:text-primary hover:text-secondary hover:bg-opacity-80 cursor-pointer hover:no-underline "
```
and then below on line 20 we can just have `anchorTagClassName + surfaceTagClassName`

Reply:
Having just played with the preview a bit, I think we should also add `text-primary` to the anchor tag in front of `visited:text-primary` so that we override the default blue on un-visited links...I think the underline and hover state are enough to indicate the link-ness, and the blue fights with the colors & makes for bad contrast in some cases. I'm open to opinions on this though!

Reply:
Yeah, that's a good point. I didn't realize this because I visited all the links already